torchmultimodal/modules/encoders/clip_text_encoder.py

```diff
@@ -39,42 +47,49 @@ def __init__(
         heads: int = 8,
         layers: int = 12,
         use_clip_init: bool = True,
+        pretrained_url: Optional[str] = None,
     ):
         super().__init__()
-        self.encoder = TransformerEncoder(
-            vocab_size=vocab_size,
-            embedding_dim=width,
-            padding_idx=-1,
-            max_seq_len=context_length,
-            num_encoder_layers=layers,
-            num_attention_heads=heads,
+        self.token_embedding = torch.nn.Embedding(vocab_size, width)
+        self.positional_embedding = torch.nn.Parameter(
+            torch.empty(context_length, width)
+        )
+        encoder_layer = TransformerEncoderLayer(
+            d_model=width,
+            nhead=heads,
             dropout=0.0,
-            normalize_before=True,
+            activation=quick_gelu,
+            norm_first=True,
         )
+        self.encoder = TransformerEncoder(encoder_layer, num_layers=layers)
 
         self.width = width
         self.context_length = context_length
 
         self.ln_final = nn.LayerNorm(width)
         self.projection = nn.Linear(width, embedding_dim, bias=False)
 
-        if use_clip_init:
+        if pretrained_url:
```

Comment:
Why not having `load_module_from_url` outside of the module so that:
-  you don't need to pass in `self` as the first argument? 
-  `pretrained` is more of an option to construct a model rather than part of the model logic

For example, in [Omnivore](https://github.com/facebookresearch/multimodal/blob/e274207e5c091d681bcb790d561fca0764148549/torchmultimodal/models/omnivore.py#L148) and [VideoCLIP](https://github.com/facebookresearch/multimodal/blob/main/examples/mugen/retrieval/video_clip.py#L163) we had `load_model` at the builder level.

Reply:
this was more for user passing in any ckpt path (vs predefined ones). but i pushed the responsibility for such things to the user for now